decode_test.go

```diff
@@ -618,6 +618,17 @@ func TestDecodeSlices(t *testing.T) {
 	}
 }
 
+func TestDecodeInterfaceSlice(t *testing.T) {
+	var v map[string]interface{}
+	if _, err := Decode(`S = []`, &v); err != nil {
```

Comment:
Could you also test nested arrays and inline tables?

```
arr = []
nested = [[]]
tbl = {arr = []}
```

Just to make sure this keeps working in all future updates!

Reply:
Actually, looking at this a bit closer the tests should be integrated with `TestDecodeSlices`, I'll just merge this as-is and fix it up; thanks!